pkg/kubectl/stop.go

```diff
@@ -335,12 +335,16 @@ func (reaper *StatefulSetReaper) Stop(namespace, name string, timeout time.Durat
 	if err != nil {
 		return err
 	}
+
 	if timeout == 0 {
 		numReplicas := ss.Spec.Replicas
+		// See discussion of this behavior here:
+		// https://github.com/kubernetes/kubernetes/pull/46468#discussion_r118589512
```

Comment:
Sounds good. So first list all pods this StS owns, ignore the ones that don't have controllerRef to this StS, and then get the sum of their `.spec.terminationGraceperiodSeconds`.